### PR TITLE
Update most_solves_in_one_competition_or_year.php to also support all time solves

### PR DESCRIPTION
**Pull request by keemy**

### CODE REVIEW

**Comment by jfly** [comment]
Your build ran into the dreaded #1629, so I've restarted it.

File: webroot/results/includes/statistics/most_solves_in_one_competition_or_year.php [hunk 1]
```diff
@@ -39,6 +39,26 @@
...
+            count(value3>0 or null)+
+            count(value4>0 or null)+
+            count(value5>0 or null) solves,
+            count(value1 and value1<>-2 or null)+
+            count(value2 and value2<>-2 or null)+
+            count(value3 and value3<>-2 or null)+
+            count(value4 and value4<>-2 or null)+
+            count(value5 and value5<>-2 or null) attempts
+  FROM      Results, Competitions competition
+  WHERE     competition.id = competitionId
```
Comment (jfly) [inline webroot/results/includes/statistics/most_solves_in_one_competition_or_year.php]: I'm surprised to see a join on Competitions here. Is this some reason for this that I'm missing?

**Comment by jfly** [comment]
The last test failure was interesting:

```
$ curl -o- -L https://yarnpkg.com/install.sh | bash
  % Total    % Received % Xferd  Average Speed   Time    Time     Time  Current
                                 Dload  Upload   Total   Spent    Left  Speed
100  5931    0  5931    0     0    197      0 --:--:--  0:00:30 --:--:--  1544
bash: line 1: syntax error near unexpected token `newline'
bash: line 1: `<!DOCTYPE html>'
The command "curl -o- -L https://yarnpkg.com/install.sh | bash" failed and exited with 2 during .
Your build has been stopped.
```

It looks like yarnpkg was down for a little while (that DOCTYPE response was coming from cloudflare). I've restarted travis yet again.

File: webroot/results/includes/statistics/most_solves_in_one_competition_or_year.php (hunk 1 above)
Comment (keemy) [inline webroot/results/includes/statistics/most_solves_in_one_competition_or_year.php]: wasn't sure but that was what the by year was doing. once i can try it on server i'll see if it's needed.

**Comment by SAuroux** [comment]
It is quite surely not needed. In the per year case, this is necessary to
group by year (an information which is not available in the Results table).
Generating the amount of allTime solves however is possible using the
Results table only.

2017-05-31 19:23 GMT+02:00 keemy <notifications@github.com>:

**Comment by jfly** [comment]
@keemy, did you get a chance to test this out on staging? Feel free to reach out to me on slack if you have questions about how to deploy this to test it out.

File: webroot/results/includes/statistics/most_solves_in_one_competition_or_year.php [hunk 2]
```diff
@@ -39,6 +39,26 @@
   LIMIT     50
 " );
 
+$allTime = polishMostSolvesAttempts( "
+  SELECT    personId,
+            "swag" whereId,
```
Comment (jonatanklosko) [inline webroot/results/includes/statistics/most_solves_in_one_competition_or_year.php]: This should be an empty string at least, right?

File: webroot/results/includes/statistics/most_solves_in_one_competition_or_year.php [hunk 3]
```diff
@@ -39,6 +39,26 @@
...
+            count(value4>0 or null)+
+            count(value5>0 or null) solves,
+            count(value1 and value1<>-2 or null)+
+            count(value2 and value2<>-2 or null)+
+            count(value3 and value3<>-2 or null)+
+            count(value4 and value4<>-2 or null)+
+            count(value5 and value5<>-2 or null) attempts
+  FROM      Results, Competitions competition
+  WHERE     competition.id = competitionId
+  GROUP BY  personId, year
```
Comment (jonatanklosko) [inline webroot/results/includes/statistics/most_solves_in_one_competition_or_year.php]: If I understand the idea we shouldn't group by year here.

**Comment by keemy** [comment]
finally got some time to finish should be good now 
image

**Comment by jonatanklosko** [comment]
Thanks! I've already deployed it and recomputed the statistics page. You can see it [there](https://www.worldcubeassociation.org/results/statistics.php#most_solves) =)

**Pull request merged**